sylvia-derive/src/contract/communication/reply.rs

```diff
@@ -475,6 +481,101 @@ impl<'a> ReplyVariant<'a> for MsgVariant<'a> {
+            let deserialized_data =
+                #sylvia ::cw_utils::parse_execute_response_data(data.as_slice())
+                    .map_err(|err| #sylvia ::cw_std::StdError::generic_err(
+                        format!("Failed deserializing protobuf data: {}", err)
+                    ))?;
+            let deserialized_data = match deserialized_data.data {
+                Some(data) => #sylvia ::cw_std::from_json(&data).map_err(|err| #sylvia ::cw_std::StdError::generic_err( #invalid_reply_data_err ))?,
+                None => return Err(Into::into( #sylvia ::cw_std::StdError::generic_err( #missing_data_err ))),
+            };
+        };
+
+        let instantiate_data_deserialization = quote! {
+            let deserialized_data =
+                #sylvia ::cw_utils::parse_instantiate_response_data(data.as_slice())
+                    .map_err(|err| #sylvia ::cw_std::StdError::generic_err(
+                        format!("Failed deserializing protobuf data: {}", err)
+                    ))?;
+        };
+
+        match data {
+            Some(data) if data.raw && data.opt => quote! {},
+            Some(data) if data.raw => quote! {
+                let data = match data {
+                    Some(data) => data,
+                    None => return Err(Into::into( #sylvia ::cw_std::StdError::generic_err( #missing_data_err ))),
+                };
+            },
+            Some(data) if data.instantiate && data.opt => quote! {
+                let data = match data {
+                    Some(data) => {
+                        #instantiate_data_deserialization
+
+                        Some(deserialized_data)
+                    },
+                    None => None,
+                };
+            },
+            Some(data) if data.instantiate => quote! {
+                let data = match data {
+                    Some(data) => {
+                        #instantiate_data_deserialization
+
+                        deserialized_data
+                    },
+                    None => return Err(Into::into( #sylvia ::cw_std::StdError::generic_err( #missing_data_err ))),
+                };
+            },
+            Some(data) if data.opt => quote! {
+                let data = match data {
+                    Some(data) => {
+                        #execute_data_deserialization
+
+                        Some(deserialized_data)
+                    },
+                    None => None,
+                };
+            },
+            None if is_data_attr => quote! {
+                let data = match data {
+                    Some(data) => {
+                        #execute_data_deserialization
+
+                        deserialized_data
+                    },
+                    None => return Err(Into::into( #sylvia ::cw_std::StdError::generic_err( #missing_data_err ))),
+                };
+            },
+            _ => {
+                emit_error!(self.name().span(), "Invalid data usage.";
```

Comment:
I am wondering if there is a way to include the error testing we have in `ui` to the cov report... We could technically do the `#[tarpaulin(ingore)]` here but this then doesn't check if the path is ever tested... Just leaving as an idea, maybe we could create issue for this, idk.

Reply:
We have an issue for this #339. Unfortunately I don't see a way to do that since `trybuild` runs its own compilations. If we would figure out something there is an issue to track that though.